src/query/api/v1/handler/prometheus/native/common.go

```diff
@@ -255,6 +264,26 @@ func parseQuery(r *http.Request) (string, error) {
 	return queries[0], nil
 }
 
+func parseLookbackDuration(r *http.Request, step time.Duration) (time.Duration, bool, error) {
+	lookback := r.FormValue(lookbackParam)
+	if lookback == "" {
+		return 0, false, nil
+	}
+
+	if lookback == stepParam {
+		// Use the step size as lookback.
+		return step, true, nil
+
+	}
+	// Otherwise it is specified as duration value.
+	value, err := parseDuration(r, lookbackParam)
+	if err != nil {
+		return 0, false, err
+	}
+
+	return value, true, nil
```

Comment:
Might have to check that `step > 0` otherwise you may have a bad time (although that case is handled elsewhere so it shouldn't panic, but might be better to catch it early)

Reply:
Sure thing will do.